mysite/code/ArticleHolder.php

```diff
@@ -59,7 +59,7 @@ public function ArchiveDates()
         $result = $query->execute();
 
         if ($result) {
-            while ($record = $result->nextRecord()) {
+            foreach ($result as $record) {
```

Comment:
`Query implements Iterator` so should be fine 👍  you happy with this @unclecheese ?

Reply:
Yup. Nice catch.